org.eclipse.xtext.common.types.tests/tests/org/eclipse/xtext/common/types/access/impl/AbstractTypeProviderTest.java

```diff
@@ -819,24 +819,46 @@ public void testMemberCount_15() {
 
 	@Test
 	public void testMemberCount_16() {
-		String typeName = TestEnum.class.getName();
-		JvmEnumerationType type = (JvmEnumerationType) getTypeProvider().findTypeByName(typeName);
-		int innerTypesCount = TestEnum.class.getDeclaredClasses().length;
-		// Nested
-		assertEquals(1, innerTypesCount);
-		int methodCount = TestEnum.class.getDeclaredMethods().length;
-		// TestEnum.values + TestEnum.valueOf
-		assertEquals(2, methodCount);
-		int constructorCount = TestEnum.class.getDeclaredConstructors().length;
-		// TestEnum(String, int, String), TestEnum(String, int, String, EnumType)
-		assertEquals(2, constructorCount);
-		int fieldCount = TestEnum.class.getDeclaredFields().length;
-		// FirstValue, SecondValue, string, ENUM$VALUES
-		assertEquals(Arrays.toString(TestEnum.class.getDeclaredFields()), 4, fieldCount);
-		// ENUM$VALUES is synthetic
-		// TestEnum(String, String, EnumType) is synthetic
-		assertEquals(type.getMembers().toString(), innerTypesCount + methodCount + constructorCount + fieldCount - 2,
-				type.getMembers().size());
+		try {
+			String typeName = TestEnum.class.getName();
```

Comment:
am not sure what our common pattern in extras is to deal with different java versions

Reply:
I don't think we do have a pattern for that.
How did you like the assumeTrue(..) thingy?

Reply:
we have the old variant. will apply it here

Reply:
problem is eclipse. the enum is compiled differently there as of now